### PR TITLE
FEAT: refactoring catalog search to support exec ed content

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
enable changes by using `features=ENABLE_EXEC_ED_INCLUSION`

related PR: https://github.com/openedx/enterprise-catalog/pull/537

https://2u-internal.atlassian.net/browse/ENT-6383
https://2u-internal.atlassian.net/browse/ENT-6381
https://2u-internal.atlassian.net/browse/ENT-6577

inclusion of exec ed in catalog search + rework of structuring order of components rendered

Base landing page
image

Search results for only exec ed, not courses or programs
image

Search results for each learning type category
image

Search results for programs and courses but not exec ed
image

Exec ed specified learning type
image

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/catalogs/CatalogSearch.jsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* eslint eqeqeq: "off" */
-import React, { useContext, useState, useMemo } from 'react';
+import React, { useContext, useState, useMemo, useEffect } from 'react';
```
Comment (kiram15) [inline src/components/catalogs/CatalogSearch.jsx]: nit: alphabetize imports

File: src/components/catalogs/CatalogSearch.jsx [hunk 2]
```diff
@@ -75,6 +99,51 @@ function CatalogSearch(intl) {
       setSelectedSuggestedCourseType('course');
     }
   };
+  
+  useEffect(() => {
+    // Grab content type(s) or default to course & program
+    let contentToDisplay = contentType ? contentType.slice() : [CONTENT_TYPE_COURSE, CONTENT_TYPE_PROGRAM];
+
+    // IFF the catalog query is a la carte, specified course type is exec ed (or course type isn't given)
```
Comment (kiram15) [inline src/components/catalogs/CatalogSearch.jsx]: nit: two f's?

Comment (kiram15) [reply src/components/catalogs/CatalogSearch.jsx]: Also this comment is confusing

File: src/constants.js [hunk 3]
```diff
@@ -27,6 +27,7 @@ export const NUM_RESULTS_PROGRAM = 4;
 export const NUM_RESULTS_COURSE = 8;
 export const COURSE_TITLE = 'Courses';
 export const PROGRAM_TITLE = 'Programs';
+export const EXEC_ED_TITLE = 'Executive Education'
```
Comment (kiram15) [inline src/constants.js]: nit: semi colon

File: src/components/catalogs/CatalogSearch.jsx [hunk 4]
```diff
@@ -52,10 +74,12 @@ function CatalogSearch(intl) {
   }
 
   let suggestedSearchContentTypeFilter;
-  if (!contentType || contentType.length === 2) {
+  if (!contentType || contentType.length === 3) {
     suggestedSearchContentTypeFilter = combinedFilter;
   } else if (specifiedContentType === CONTENT_TYPE_PROGRAM) {
     suggestedSearchContentTypeFilter = programFilter;
+  } else if (specifiedContentType === CONTENT_TYPE_PROGRAM) {
```
Comment (kiram15) [inline src/components/catalogs/CatalogSearch.jsx]: This is more just a point of clarification for me, aren't exec ed courses and not programs? also this if is identical to the one above it so you could just combine them if thats what you're trying to do. If not, this needs to be fixed

File: src/components/catalogs/CatalogSearch.jsx [hunk 5]
```diff
@@ -75,6 +99,51 @@ function CatalogSearch(intl) {
       setSelectedSuggestedCourseType('course');
     }
   };
+  
+  useEffect(() => {
+    // Grab content type(s) or default to course & program
+    let contentToDisplay = contentType ? contentType.slice() : [CONTENT_TYPE_COURSE, CONTENT_TYPE_PROGRAM];
```
Comment (marlonkeating) [inline src/components/catalogs/CatalogSearch.jsx]: nit: While slice() is a perfectly valid way to make a shallow copy, my general preference is to use an operator that does that specific operation, rather than a function built to do a different operation (but will do the operation you want if you omit certain arguments). 
```suggestion
    let contentToDisplay = contentType ? [...contentType] : [CONTENT_TYPE_COURSE, CONTENT_TYPE_PROGRAM];
```

File: src/components/catalogs/CatalogSearch.jsx (hunk 2 above)
Comment (alex-sheehan-edx) [reply src/components/catalogs/CatalogSearch.jsx]: IFF stands for "If and only if"

File: src/constants.js (hunk 3 above)
Comment (alex-sheehan-edx) [reply src/constants.js]: 🤦

File: src/components/catalogs/CatalogSearch.jsx (hunk 5 above)
Comment (alex-sheehan-edx) [reply src/components/catalogs/CatalogSearch.jsx]: love this!

File: src/components/catalogSearchResults/CatalogSearchResults.jsx [hunk 6]
```diff
@@ -319,9 +388,13 @@ export function BaseCatalogSearchResults({
     );
   }
 
+  function CustomRowStatus() {
+    return null;
+  }
```
Comment (marlonkeating) [inline src/components/catalogSearchResults/CatalogSearchResults.jsx:408]: clarification: What's the significance of 'null' for CustomRowStatus?  (As opposed to 'undefined' which is the other value used in line 458)

File: src/components/catalogSearchResults/CatalogSearchResults.jsx [hunk 7]
```diff
@@ -260,7 +323,12 @@ export function BaseCatalogSearchResults({
     };
 
   function contentTitle() {
-    let subTitle = contentType === CONTENT_TYPE_COURSE ? COURSE_TITLE : PROGRAM_TITLE;
+    let subTitle;
```
Comment (alex-sheehan-edx) [inline src/components/catalogSearchResults/CatalogSearchResults.jsx:337]: there's def a better way to do this

**Review by marlonkeating — COMMENTED** [review]
I Don't see any large issues, but there's a few things I think we should consider as part of a future refactor (perhaps when we remove the flag gating).

To someone new to the code, it's quite hard to follow a lot of the logic between the particulars of the data table component, Algolia queries, and all the bits of filter logic (+ feature gating).  I think some additional compartmentalization and abstraction could greatly help readability and reusability.

File: src/components/catalogSearchResults/CatalogSearchResults.jsx (hunk 6 above)
Comment (alex-sheehan-edx) [reply src/components/catalogSearchResults/CatalogSearchResults.jsx:408]: oh uhhh, I figure undefined is preferable here

File: src/components/catalogSearchResults/CatalogSearchResults.jsx [hunk 8]
```diff
@@ -208,6 +234,33 @@ export function BaseCatalogSearchResults({
...
+        Cell: TitleButtonComponent,
+      },
+      {
+        Header: TABLE_HEADERS.partner,
+        accessor: 'partners[0].name',
+      },
+      {
+        Header: TABLE_HEADERS.price,
+        accessor: 'entitlements',
+        Cell: ({ row }) => (row.values.entitlements[0].price
```
Comment (kiram15) [inline src/components/catalogSearchResults/CatalogSearchResults.jsx:251]: The column version properly has price but the cards dont?

File: src/components/catalogSearchResults/CatalogSearchResults.jsx (hunk 7 above)
Comment (kiram15) [reply src/components/catalogSearchResults/CatalogSearchResults.jsx:337]: Honestly I would break it up into another if else, this way makes sense but its kind of confusing even if its less lines

**Review by kiram15 — CHANGES_REQUESTED** [review]
Connected w/ Alex offine because there are some problems with some of the courses and programs not showing up as they should, which means changes need to be made before this can be merged in.

File: src/components/catalogSearchResults/CatalogSearchResults.jsx (hunk 8 above)
Comment (alex-sheehan-edx) [reply src/components/catalogSearchResults/CatalogSearchResults.jsx:251]: I just forgot to update the cards 🙈

File: src/components/catalogSearchResults/CatalogSearchResults.jsx (hunk 6 above)
Comment (alex-sheehan-edx) [reply src/components/catalogSearchResults/CatalogSearchResults.jsx:408]: actually, I'm taking that back, I believe this needs to be null.

**Review by kiram15 — APPROVED** [review]
LGTM!
